### PR TITLE
TF2

### PR DESCRIPTION
**Pull request by jdeschamps**
So I think I did the bare minimum to move DenoiSeg to TF2:
* Updated the conda environment instructions to those of N2V
* Updated setup.py
* Changed Keras calls to tensorflow.keras
* Adapted some part of the code based on similar N2V code
* Ran all three notebooks examples

Before merging, there are some questions left hanging:
* Is Keras still necessary? (conda instructions and setup.py)
* DenoiSeg now uses a layer from tensorflow.compat.v1.nn (softmax_cross_entropy_with_logits_v2, see losses.py), it would probably be better to change it to something that will not be deprecated in the future
* I do not have a compatible GPU so I could not test the performances, just that it ran in the notebooks and that few epochs (with very small batches) still gave something alright
* In order to compile setup.py I had to fix some package versions required by TF 2.4.1 ("six<=1.15.0", "typing-extensions<=3.7.4", "wrapt<=1.12.1") that were not fulfilled by conda dependency solver. I don't know if that was the best decision in a Python context (lack of personal experience).

It would be great if you could have a look!

### CODE REVIEW

**Comment by jdeschamps** [comment]
@tibuch :D

**Comment by tibuch** [comment]
Everything is running, thanks @jdeschamps.

But I found a few things, which would be nice to fix:
* Datasets are currently limited, which leads to problems if steps_per_epoch x batch_size is larger than the dataset length. As long as steps_per_epoch x batch_size is smaller than len(dataset) everything works fine.
* The image segmentation tensorboard is currently not working and commented. The normal loss tensorboard is available.
* The first epoch usually takes super long. In general it feels not very performant. 

I think we can merge this and document the above as issues.

**Review by tibuch — APPROVED** [review]

**Comment by jdeschamps** [comment]
Thanks! I'll create the issues.
